lib/resty/healthcheck.lua

```diff
@@ -281,20 +282,25 @@ end
 function checker:add_target(ip, port, hostname, is_healthy, hostheader)
   ip = tostring(assert(ip, "no ip address provided"))
   port = assert(tonumber(port), "no port number provided")
+  hostname = hostname or ip
```

Comment:
Mostly just curious: I saw this nil hostname handling logic was added in several other places. Is there a reason why?

Reply:
It was a mistake, thanks for pointing that out. 

We keep the health status in a table indexed by ip, port and hostname. When there's no hostname set, it's ip, port and ip again. It was just a matter of never letting the hostname being empty in the line you pointed out, but I spread this "fix" everywhere. It was already reverted.